pkg/util/kafka/common.go

```diff
@@ -166,3 +171,41 @@ func newListenerSecurityProtocolMap(s string) listenerSecurityProtocolMap {
 
 	return listenerSecProtoMap
 }
+
+func GetBootstrapServers(cr *v1beta1.KafkaCluster) (bootstrapServers string) {
+	var listenerName string
+	var listenerServiceName string
+	var listenerStatusList v1beta1.ListenerStatusList
+
+	for _, lc := range cr.Spec.ListenersConfig.InternalListeners {
+		if lc.UsedForInnerBrokerCommunication && !lc.UsedForControllerCommunication {
+			listenerName = lc.Name
+		}
+	}
+
+	listenerStatusList, ok := cr.Status.ListenerStatuses.InternalListeners[listenerName]
+	if !ok {
+		return
+	}
+
+	switch {
+	case cr.Spec.HeadlessServiceEnabled:
+		listenerServiceName = HeadlessListenerServiceName
+	default:
+		listenerServiceName = AnyBrokerListenerServiceName
+	}
```

Comment:
I think an If would be more appropriate here instead of a switch case with 1 case

Reply:
Agreed. Will refactor.